website_compassion/controllers/my_account.py

```diff
@@ -270,24 +272,43 @@ def my_child(self, state="active", child_id=None, **kwargs):
             return request.redirect(f"/my/children?child_id={children[0].id}")
 
     @route("/my/donations", type="http", auth="user", website=True)
-    def my_donations(self, form_id=None, **kw):
+    def my_donations(self, invoice_page='1', form_id=None, **kw):
         """
         The route to the donations and invoicing page
+        :param invoice_page: index of the invoice pagination
         :param form_id: the id of the filled form or None
         :param kw: additional optional arguments
         :return: a redirection to a webpage
         """
         partner = request.env.user.partner_id
 
-        invoices = request.env["account.invoice"].sudo().search([
+        invoice_page = int(invoice_page) if invoice_page.isnumeric() else 1
+
+        invoice_search_criteria = [
             ("partner_id", "=", partner.id),
             ("state", "=", "paid"),
             ("invoice_category", "in", ["sponsorship", "fund", "other"]),
             ("type", "=", "out_invoice"),
             ("amount_total", "!=", 0),
-        ])
+        ]
+
+        sudo_invoice_env = request.env["account.invoice"].sudo()
+
+        all_invoice_count = sudo_invoice_env.search_count(invoice_search_criteria)
+
+        invoice_per_page = 30
```

Comment:
Suggestion: maybe the invoice_per_page can be a parameter of the function with default = 30